content/articles/diogogpinto-filament-themes-full-guide.md

```diff
@@ -0,0 +1,333 @@
+
+```css
+@import '/vendor/filament/filament/resources/css/theme.css';
+@import '/node_modules/cropperjs/dist/cropper.css';
+@import '/vendor/awcodes/filament-curator/resources/css/plugin.css';
+
+@config 'tailwind.config.js';
+```
+
+This last step imports all custom classes needed by the plugin to render correctly. It imports the class, puts into our panel's unique CSS file and minimizes it all for greater performance.
+
+#### Running the build process
+
+When installing plugins and editing any theme related files, always remember to run `npm run build` when you're finished to compile all the assets.
+
+## Customizing the theme
+
+Laravel, and Filament being Laravel based, offer multiple solutions to the same problem. So, we will address two ways of building your own Filament look and feel, the recommended way and the not-so-recommended way. Let's start with the latter.
+
+### Before starting the customization process
+
+You should start the `npm run dev` process in the root of your project. This process watches for changes in the files registered in the `vite.config.js` (and `tailwind.config.js` by extension) and automatically builds a temporary CSS file to render the changes on file save.
+
+> [!note]
+> You still need to run `npm run build` when you're done customizing your theme
+
+### The recommended way to customize your theme
+
+The Filament panels core offers you a list of classes. Every component of the panel layout, including forms, infolists and tables, were wrapped in custom classes to make it possible, for any developer, to customize the panel look and feel. Let's look at the following example:
+
+![Breadcrumb Example](/images/content/articles/diogogpinto-filament-themes-full-guide/breadcrumb-example.webp)
+
+Looking at the breadcrumb component with Chrome's builtin Developer Tools > Inspect Element, you can see that the `<nav>`na tag contains the following classes:
+
+```html
+<nav class="fi-breadcrumbs mb-2 hidden sm:block">
+```
+
+The `.fi-breadcrumbs` class is inserted to allow developers to further optimize the breadcrumbs' look and feel. By default, the class has no CSS properties defined in itself, it's only there for the developer to apply custom styles.
+
+#### Customizing the CSS
+
+Picking up in the above example of `.fi-breadcrumbs`, we can now apply custom CSS classes on our `theme.css`. You can either use Tailwind utility classes or just write vanilla CSS. Let's say you want to override the default margin `mb-2`. Inputting the below CSS will yield the exact same results:
+
+```css
+.fi-breadcrumbs {
+    @apply mb-4;
+}
+
+.fi-breadcrumbs {
+    margin-bottom: 1rem;
+}
+```
+
+As the `.fi-breadcrumbs` class is inserted before the default Tailwind utility classes, applying custom CSS will always override the default style.
```

Comment:
See above comment, but I believe the order in which a class string is laid out has no bearing on specificity or the cascade, so having multiple classes _of the same specificity_ may lead to unreliable outputs based on where the Tailwind classes are defined in relation to one another.

Reply:
Yup, totally depends on specificity.

Reply:
@zepfietje @alexandersix 

I'm probably missing something, given that both TW classes and custom classes created with @apply are technically single class selectors, shouldn’t they all have equal specificity?

I clearly made a mistake in the text, as it isn't order of declaration that matters, but the build process that builds custom classes after the utility classes.

Hope to get some clarity, thank you in advance!

Reply:
Do you think this makes sense?